docs/general/user-defined-functions.rst

```diff
@@ -242,39 +259,39 @@ function must return a ``double precision`` array of size 2, ``WKT`` string or
 Here is an example of a JavaScript function returning a ``double array``::
 
     cr> CREATE FUNCTION rotate_point(point geo_point, angle real)
-    ...  RETURNS geo_point
-    ...  LANGUAGE JAVASCRIPT
-    ...  AS 'function rotate_point(point, angle) {
-    ...        var cos = Math.cos(angle);
-    ...        var sin = Math.sin(angle);
-    ...        var x = cos * point[0] - sin * point[1];
-    ...        var y = sin * point[0] + cos * point[1];
-    ...        return [x, y];
-    ...      }';
+    ... RETURNS geo_point
+    ... LANGUAGE JAVASCRIPT
+    ... AS 'function rotate_point(point, angle) {
+    ...       var cos = Math.cos(angle);
+    ...       var sin = Math.sin(angle);
+    ...       var x = cos * point[0] - sin * point[1];
+    ...       var y = sin * point[0] + cos * point[1];
+    ...       return [x, y];
+    ...    }';
     CREATE OK, 1 row affected  (... sec)
 
 Below is an example of a JavaScript function returning a ``WKT`` string, which
 will be cast to ``geo_point``::
 
      cr> CREATE FUNCTION symmetric_point(point geo_point)
-     ...  RETURNS geo_point
-     ...  LANGUAGE JAVASCRIPT
-     ...  AS 'function symmetric_point (point, angle) {
-     ...        var x = - point[0],
-     ...            y = - point[1];
-     ...        return "POINT (\" + x + \", \" + y +\")";
-     ...      }';
+     ... RETURNS geo_point
+     ... LANGUAGE JAVASCRIPT
+     ... AS 'function symmetric_point (point, angle) {
+     ...       var x = - point[0],
+     ...           y = - point[1];
+     ...       return "POINT (\" + x + \", \" + y +\")";
+     ...    }';
      CREATE OK, 1 row affected  (... sec)
 
 Similarly, if the function specifies the ``geo_shape`` return data type, then
-the JavaScript function should return a ``GeoJson`` object or``WKT`` string::
+the JavaScript function should return a ``GeoJson`` object or ``WKT`` string::
 
-     cr> CREATE FUNCTION line(start_point array(double precision), end_point array(double precision))
-     ...  RETURNS object
-     ...  LANGUAGE JAVASCRIPT
-     ...  AS 'function line(start_point, end_point) {
+     cr> CREATE FUNCTION line("start" array(double precision), "end" array(double precision))
+     ... RETURNS object
+     ... LANGUAGE JAVASCRIPT
+     ... AS 'function line(start, end) {
```

Comment:
Would the old definition still work, or do users need to update their function definitions?

Reply:
Function definitions stay the same. I updated it in the example to make it a bit more concise. There shouldn't be any issues with backward compatibility, such as we validate, register and compile functions on a startup by getting its definitions it from metadata. I'll double check it.

Don't have some presentable results, but will provide it a bit later before we merge this PR.